src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/UnsupportedTypeConverter.cs

```diff
@@ -5,10 +5,16 @@ namespace System.Text.Json.Serialization.Converters
 {
     internal sealed class UnsupportedTypeConverter<T> : JsonConverter<T>
     {
+        private readonly string? _errorMessage;
+
+        public UnsupportedTypeConverter(string? errorMessage = null) => _errorMessage = errorMessage;
+
+        public string ErrorMessage => _errorMessage ?? SR.Format(SR.SerializeTypeInstanceNotSupported, typeof(T).FullName);
```

Comment:
it might make sense to `??=` here and remove `readonly` from `_errorMessage`

Reply:
It's intentional. I wouldn't want the converter to root additional memory just to optimize throwing exceptions.